test/integration/TypeBrowserTest.kt

```diff
@@ -16,116 +16,133 @@
+                clickIcon(composeRule, Icon.Code.CHEVRONS_UP)
+                delayAndRecompose(composeRule)
 
-            composeRule.onNodeWithText("commit-date").assertDoesNotExist()
+                nodeWithTextDoesNotExist(composeRule, "commit-date")
 
-            composeRule.onNodeWithText(DOUBLE_CHEVRON_DOWN_ICON_STRING).performClick()
-            delayAndRecompose(composeRule)
+                clickIcon(composeRule, Icon.Code.CHEVRONS_DOWN)
+                delayAndRecompose(composeRule)
 
-            composeRule.onNodeWithText("commit-date").assertExists()
+                nodeWithTextExists(composeRule, "commit-date")
+            }
         }
     }
 
     // This test is ignored as the export schema button doesn't open a new file during testing.
     @Ignore
     @Test
     fun exportSchema() {
-        studioTestWithRunner(composeRule) { address ->
-            connectToTypeDB(composeRule, address)
-            cloneAndOpenProject(composeRule, source = TQL_DATA_PATH, destination = testID)
-            createDatabase(composeRule, dbName = testID)
-            writeSchemaInteractively(composeRule, dbName = testID, SCHEMA_FILE_NAME)
-
-            StudioState.client.session.tryOpen(database = testID, TypeDBSession.Type.DATA)
-            delayAndRecompose(composeRule, Delays.NETWORK_IO)
-
-            StudioState.schema.exportTypeSchema { schema ->
-                StudioState.project.current!!.reloadEntries()
-                StudioState.project.tryCreateUntitledFile()?.let { file ->
-                    file.content(schema)
-                    file.tryOpen()
+        withTypeDB { typeDB ->
+            runBlocking {
+                connectToTypeDB(composeRule, typeDB.address())
+                createData(source = TQL_DATA_PATH, destination = testID)
+                openProject(composeRule, testID)
+                createDatabase(composeRule, dbName = testID)
+                writeSchemaInteractively(composeRule, dbName = testID, SCHEMA_FILE_NAME)
+
+                StudioState.client.session.tryOpen(database = testID, TypeDBSession.Type.DATA)
+                delayAndRecompose(composeRule, Delays.NETWORK_IO)
+
+                StudioState.schema.exportTypeSchema { schema ->
+                    StudioState.project.current!!.reloadEntries()
+                    StudioState.project.tryCreateUntitledFile()?.let { file ->
+                        file.content(schema)
+                        file.tryOpen()
+                    }
                 }
-            }
-            composeRule.waitForIdle()
+                composeRule.waitForIdle()
 
-            composeRule.onNodeWithText("define").assertExists()
-            composeRule.onNodeWithText("# This program is free software: you can redistribute it and/or modify").assertDoesNotExist()
+                nodeWithTextExists(composeRule, "define")
+                nodeWithTextDoesNotExist(composeRule, "# This program is free software: you can redistribute it and/or modify")
```

Comment:
By the way, what is this for?

Reply:
This whole test is ignored, but it is now becoming a source of confusion so I'm scrapping this test as I haven't been able to get the export schema button to work as expected (open a new file with the schema for the database open within it).

The highlighted section of code aims to assert that the schema generated by the open schema button is the one TypeDB has passed back to us rather than the schema we just opened to write to the database.